src/Stratis.Bitcoin.Features.BlockStore/AddressIndexing/AddressIndexer.cs

```diff
@@ -128,11 +132,13 @@ public AddressIndexer(StoreSettings storeSettings, DataFolder dataFolder, ILogge
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
 
             this.averageTimePerBlock = new AverageCalculator(200);
+            this.maxReorg = this.network.Consensus.MaxReorgLength == 0 ? (int)this.network.Consensus.MaxReorgLength : FallBackMaxReorg;
         }
 
         public void Initialize()
         {
-            if (!this.storeSettings.AddressIndex)
+            // The transaction index is needed in the event of a reorg.
+            if (!this.storeSettings.AddressIndex || !this.storeSettings.TxIndex)
```

Comment:
why txindex is needed?

Reply:
It's been added and removed from the indexer several times, this slipped through the last removal. Fixed.